testsuite/Properties.hs

```diff
@@ -0,0 +1,30 @@
+module Main (
+    main
+  ) where
+
+import Test.Tasty
+import Test.Tasty.QuickCheck as QC
+
+import Morte.Core
+import ClosedWellTyped
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Properties" [
+      QC.testProperty "normalize . normalize == normalize" normalizeIdempotent 
+    , QC.testProperty "Well typed after normalization"     wellTypedAfterNormalize
```

Comment:
Yeah, I checked this but decided not to included untill we talked about it a little bit more.
I will include it in the update but instead of fmap-ing into the ``Either`` result I think it's better to use implication as in the second test because ``typeOf`` will always be a ``Right`` with the generator and using ``fmap`` will further require an ``Eq`` constraint on ``TypeError`` and ``TypeMessage``.